internal/cmd/asyncapi/command_import.go

```diff
@@ -0,0 +1,444 @@
+	if err != nil {
+		return nil, err
+	}
+
+	flags := &flagsImport{
+		overwrite:               overwrite,
+		kafkaApiKey:             kafkaApiKey,
+		schemaRegistryApiKey:    schemaRegistryApiKey,
+		schemaRegistryApiSecret: schemaRegistryApiSecret,
+	}
+	return flags, nil
+}
+
+func (c *command) parse(cmd *cobra.Command, args []string) error {
+	spec, err := fileToStruct(args[0])
+	if err != nil {
+		return err
+	}
+	// Get flags
+	flagsImp, err := getFlagsImport(cmd)
+	if err != nil {
+		return err
+	}
+	// Getting Kafka Cluster & SR
+	flags := createExportFlagsFromImportFlags(flagsImp)
+	details, err := c.getAccountDetails(&flags)
+	if err != nil {
+		return err
+	}
+	for topicName, topicDetails := range spec.Channels {
+		err = c.addChannelToCluster(details, spec, topicName, topicDetails.TopicBinding.Kafka, false, flagsImp.overwrite)
+		if err != nil {
+			if err.Error() == parseErrorMessage {
+				log.CliLogger.Info(err)
+			} else {
+				log.CliLogger.Warn(err)
+			}
+		}
+	}
+	output.Printf("AsyncAPI specification imported from \"%s\"\n", args[0])
+	return nil
+}
+
+func fileToStruct(fileName string) (*Spec, error) {
+	asyncSpec, err := os.ReadFile(fileName)
+	if err != nil {
+		return nil, err
+	}
+	spec := new(Spec)
+	err = yaml.Unmarshal(asyncSpec, &spec)
+	if err != nil {
+		return nil, fmt.Errorf("unable to unmarshal yaml file: %v", err)
+	}
+	return spec, nil
+}
+
+func (c *command) addChannelToCluster(details *accountDetails, spec *Spec, topicName string, kafkaBinding KafkaBinding, topicNotCreated, overwrite bool) error {
+	output.Printf("Importing topic: %s\n", topicName)
+	topicExists, err := c.addTopic(details, topicName, spec.Channels[topicName].Description, kafkaBinding, overwrite)
+	if err != nil {
+		// unable to create kafka topic
+		log.CliLogger.Warn(err)
+		topicNotCreated = true
+	}
+	// If topic exists and overwrite flag is false, move to the next channel in spec
+	if topicExists && !overwrite {
+		return errors.New(parseErrorMessage)
+	}
+	// Register schema
+	schemaId, err := registerSchema(details, topicName, spec.Components)
+	if err != nil {
+		return err
+	}
+	// Update subject compatibility
+	if err := updateSubjectCompatibility(details, spec.Channels[topicName].XMessageCompatibility, topicName+"-value"); err != nil {
+		return err
+	}
+	// Add tags
+	output.Println("Adding tags at schema level and topic level.")
+	if _, _, err := addSchemaTags(details, spec.Components, topicName, schemaId); err != nil {
+		return err
+	}
+	if !topicNotCreated && spec.Channels != nil {
+		if _, _, err := addTopicTags(details, spec.Channels[topicName].Subscribe, topicName); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func createExportFlagsFromImportFlags(flagsImp *flagsImport) flags {
+	flagsExport := flags{
+		kafkaApiKey:             flagsImp.kafkaApiKey,
+		schemaRegistryApiKey:    flagsImp.schemaRegistryApiKey,
+		schemaRegistryApiSecret: flagsImp.schemaRegistryApiSecret,
+	}
+	return flagsExport
```

Comment:
What is this function accomplishing? Do we need it?
```suggestion
	return flags{
		kafkaApiKey:             flagsImp.kafkaApiKey,
		schemaRegistryApiKey:    flagsImp.schemaRegistryApiKey,
		schemaRegistryApiSecret: flagsImp.schemaRegistryApiSecret,
	}
```

Reply:
we are calling getAccountDetails function which requires `flag` struct object. `flagsImport` and `flags` are two different structs. I can put it in the parse function itself.